File: php/PC.php
<?php

class PC
{
    private $procObj;
    private $diskObj;
    private $ramObj;
    private $totalPrice;

    # Constructor
    public function __construct(Processor $p, Disk $d, Ram $r)
    {
        $this->procObj = $p;
        $this->diskObj = $d;
        $this->ramObj = $r;
        $this->totalPrice = $p->getPrice() + $d->getPrice() + $r->getPrice();
    }

    # Setter and Getter
    public function setProcessor(Processor $p)
    {
        $this->procObj = $p;
    }

    public function getProcessor()
    {
        return $this->procObj;
    }

    public function setDisk(Disk $d)
    {
        $this->diskObj = $d;
    }

    public function getDisk()
    {
        return $this->diskObj;
    }

    public function setRam(Ram $r)
    {
        $this->ramObj = $r;
    }

    public function getRam()
    {
        return $this->ramObj;
    }

    public function setTotalPrice()
    {
        $this->totalPrice = $this->procObj->getPrice() + $this->diskObj->getPrice() + $this->ramObj->getPrice();
    }

    public function getTotalPrice()
    {
        return $this->totalPrice;
    }

    # Menampilkan isi kelas
    public function showPC()
    {
        $this->procObj->showProcessor();
        $this->diskObj->showDisk();
        $this->ramObj->showRam();
        echo "Total Price     : Rp. " . $this->totalPrice . "<br>";
    }
}

File: c++/PC.cpp
#include "Processor.cpp"
#include "Disk.cpp"
#include "Ram.cpp"

class PC
{
    private:
        /* atribute */
        int totalPrice;
        Processor procObj;
        Disk diskObj;
        Ram ramObj;
    public:
        PC(){
        }

        PC(Processor p, Disk d, Ram r){
            this->procObj = p;
            this->diskObj = d;
            this->ramObj = r;
            this->totalPrice = procObj.getProcessorPrice() + diskObj.getDiskPrice() + ramObj.getRamPrice();
        }

        //setter
        void setProcessor(Processor p){
            this->procObj = p;
        }
        void setDisk(Disk d){
            this->diskObj = d;
        }
        void setRam(Ram r){
            this->ramObj = r;
        }
        void setTotalPrice(){
            this->totalPrice = procObj.getProcessorPrice() + diskObj.getDiskPrice() + ramObj.getRamPrice();
        }

        //getter
        Processor getProcessor(){
            return this->procObj;
        }
        Disk getDisk(){
            return this->diskObj;
        }
        Ram getRam(){
            return this->ramObj;
        }
        int getTotalPrice(){
            return this->totalPrice;
        }

        //print
        void showPC(){
            procObj.showProcessor();
            diskObj.showDisk();
            ramObj.showRam();
            cout << "Total Price     : Rp. " << totalPrice << endl;
        }

        ~PC(){}
};